components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationIdentityProviderMgtListener.java

```diff
@@ -55,144 +48,129 @@ public boolean doPreUpdateIdP(String oldIdPName, IdentityProvider identityProvid
             IdentityProviderManagementException {
 
         try {
-            IdentityServiceProviderCache.getInstance().clear(tenantDomain);
-
             IdentityProviderManager identityProviderManager = IdentityProviderManager.getInstance();
 
-            ConnectedAppsResult connectedApplications;
-            String idpId = identityProviderManager.getIdPByName(oldIdPName, tenantDomain).getResourceId();
+            IdentityProvider oldIdentityProvider = identityProviderManager.getIdPByName(oldIdPName, tenantDomain);
```

Comment:
Can this be null?

Reply:
```bash
String idpId = identityProviderManager.getIdPByName(oldIdPName, tenantDomain).getResourceId();
```
Previously, we used the above logic to read the resource ID. I believe it shouldn't be null, as it would have caused a failure before if it were.